### ISSUE TITLE
Can it be used within Angular 2

### ISSUE THREAD

**Issue by gertadriaensen** [minhur/bootstrap-toggle#116]
I have a problem that I can not get this working within Angular 2.
I have added the line "bootstrap-toggle": "2.2.2" in the packege.sjon file and the node_modules are added within the structure.

I also just see a regular checkbox in my HTML and not the fancy bootstrap. Anyone a suggestion for this? 
<input type="checkbox" checked data-toggle="toggle">

Also the bootstraptoggle css is not applied but I do not directly see why that is.

I am very new to Angular 2 and never used AngularJS so that is a disadvantage as well.

**Comment by rmoscowitz** [comment]
Yes, it can be done... although I'm interested if anyone has figured out a better way. I did something like this:

``` ts
import 'bootstrap-toggle';
import 'bootstrap-toggle/css/bootstrap-toggle.min.css';
import {AfterViewInit, Component} from '@angular/core';

const template = `
  <div>
    <input id="my-toggle"
           type="checkbox" 
           [(ngModel)]="toggleValue"
           data-toggle="toggle" 
           data-on="ON!!!" 
           data-off="OFF!!!">
  </div>
`;

@Component({
  selector: 'my-component',
  template
})
export class MyComponent implements AfterViewInit {

  toggleValue:boolean = true;

  constructor() {
  }

  ngAfterViewInit() {
    $('#my-toggle').bootstrapToggle();
    $('#my-toggle').change((event) => {
      this.toggleValueChanged(event.target.checked);
    });
  }

  toggleValueChanged(toggleValue:boolean) {
    console.log('toggle changed', toggleValue);
    this.toggleValue = toggleValue;
  }
}
```

I struggled a little bit because I was not able to bind directly to the `change` event in my template, which you can do with a simple checkbox. You also can't directly bind to the bootstrap events. Also, I wanted to refer to the checkbox as a ViewChild, but couldn't call `bootstrapToggle` without wrapping it in a jquery object anyway (so I just stayed with querying for the element id).

**Comment by MikeAirey** [comment]
@rmoscowitz Thanks, this did the trick nicely!
I used an Angular2 template reference variable & ViewChild in place of the DOM ID, which also seems to work fine. While I was prototyping the change detector was not picking up model changes from the bootstrapToggle, so I had to inject ChangeDetectorRef and call detectChanges() in toggleValueChanged(). But since dropping it in a wrapper component which exports the changing value, the change detector picks up the changes normally.
